composer.json

```diff
@@ -10,7 +10,7 @@
         }
     ],
     "require": {
-        "php": "^7.2 || 8.0.*"
+        "php": ">=7.2"
```

Comment:
wouldn't it be better to go for `^7.2 || ^8.0` ?

Reply:
Not really, as per community understanding of this being best practice using >= op
See symfony discussion.
Everything else is - as this PR is proof of - an unnecessary burden on all stakeholders.

Reply:
Well, it breaks it for 1% of the use cases usually.
So very likely there is nothing that will directly stop working from this lib then :)

Reply:
Refs https://github.com/symfony/symfony/blob/5.4/composer.json#L35 etc

Reply:
@dereuromark I did not follow the discussion at Symfony, if there was one, but using '>= 7.x' means that this version of this package will be seen as useable when running composer on php 9.x, which will most likely break for more than 1% of usecases.
Also, adding a new minor version $current.1 where you limit the supported php version, after you find out that version $current does in fact not work with php version $new, as version $current is already published, and will be found by composer and deemed useable when users start using php $new - the only workaround I can think about that is users declaring in their end apps the incompatibility with lib.$current.

I have been hit by a case of this very issue this morning, as I have a pipeline using sami instead of phpdoc, which is falling back to using sami 3 when running on php 8, instead of failing as it should (sami 3 has a php>=5.3 constraint, whereas sami 4 has php^7.1)...

Reply:
Sure, it has other drawbacks.
But usually the drawbacks of not using the permissive strategy are worse in less maintained repos and their cascading dependency chain.
Impossible usage despite perfectly fine to work together.

In the end I dont care though what we use here. As long as we are out of the deadlock we are currently facing everywhere because of this single dependency :)

Reply:
I adjusted it as per your comment for now to move on.